DirectProgramming/DPC++FPGA/Tutorials/Features/ac_types/ac_int/README.md

```diff
@@ -0,0 +1,250 @@
+| What you will learn               | Using the ac_int data-type for basic operations <br> Efficiently using the left shift operation <br> Setting and reading certain bits of an ac_int number
+| Time to complete                  | 20 minutes
+
+
+
+## Purpose
+
+This FPGA tutorial shows how to use the ac_int type with some simple examples.
+
+This data-type can be used in place of native integer types to generate area efficient and optimized designs for the FPGA. For example, operations which do not utilize all of the bits the native integer types are good candidates for replacement with ac_int type.
+
+
+### Simple Code Example
+
+An ac_int number can be defined as follows:
+```cpp
+ac_int<W, S> a;
+```
+Here W is the width and S is the sign of the number. Signed numbers use one of the W bits to store the sign information.
+
+To use this type in your code, you must include the following header:
+
+```cpp
+#include <sycl/ext/intel/ac_types/ac_int.hpp>
+```
+Additionally, you must use the flag `-qactypes` in order to ensure that the headers are correctly included.
+
+For convenience, the following are predefined under the `ac_intN` namespace:
+```
+ac_int<N, true>  are type defined as intN up to 63.
+ac_int<N, false> are type defined as uintN up to 63.
+```
+
+For example, a 14 bit signed ac_int can be defined by using
+```cpp
+ac_intN::int14 a;
+```
+
+### Understanding the Tutorial Design
+
+The tutorial consists of several functions, each of which contains a SYCL kernel that demonstrates a specific operation. The operations we will see are:
+* Addition
+* Division
+* Multiplication
+* Left shift
+* Setting a bit of an ac_int number
+* Reading a bit of an ac_int number
+
+#### Basic Operations and Promotion Rules
+
+When using ac_int, we can write Addition, Division, Multiplication operations to use precisely as many bits as are needed to store the results. This is demonstrated by the kernels `Add`, `Div` and `Mult`.
+
+ac_int automatically promotes the result of all operations to the number of bits needed to represent all possible results without overflowing. For example, the addition of two 8-bit integers results in a 9-bit result to support overflow.
+
+However, if the user attempts to store the result in an 8-bit container, ac_int will let the user do this, but this leads to the discard of the extra carry bit. The responsibility lies on the user to use the correct datatype.
+
+These promotions rules are consistent across all architectures so the behavior should be equivalent on x86 or on FPGA.
+
+#### Shift Operation
+
+The behavior of a shift operation with ac_ints is slightly different from its behvior with native integer types. For full details, see the ac_int documentation in the file `ac_data_types_ref.pdf`. Some key points to remember are as follows:
+- If the datatype of the shift amount is signed, negative shift amounts are accepted. A shift by a negative amount is equivalent to a positive shift in the opposite direction.
```

Comment:
Passive voice. try writing as:

If the datatype of the shift amount is not explicitly unsigned (either using `ac_int<N, *false*>` or using the `unsigned` keyword), then the compiler will generate a more complex shifter that allows negative shifts and positive shifts. A right-shift by a negative amount is equivalent to a positive left-shift.